tests/queries/0_stateless/02668_parse_datetime_in_joda_syntax.sql

```diff
@@ -60,7 +60,7 @@ select parseDateTimeInJodaSyntax('+99 02', 'xx ww', 'UTC') = toDateTime('1999-01
 select parseDateTimeInJodaSyntax('10 +10', 'ww xx', 'UTC') = toDateTime('2010-03-08', 'UTC');
 select parseDateTimeInJodaSyntax('2+10', 'wwxx', 'UTC') = toDateTime('2010-01-11', 'UTC');
 select parseDateTimeInJodaSyntax('+102', 'xxM', 'UTC') = toDateTime('2010-02-01', 'UTC');
-select parseDateTimeInJodaSyntax('+20102', 'xxxxM', 'UTC') = toDateTime('2010-02-01', 'UTC');
+select parseDateTimeInJodaSyntax('+20102', 'xxxxM', 'UTC') = toDateTime('2010-02-01', 'UTC'); -- { serverError CANNOT_PARSE_DATETIME }
```

Comment:
Why did it become an error? This looks like a backward incompatible change to me

Reply:
According to https://docs.oracle.com/javase/8/docs/api/java/text/SimpleDateFormat.html#year, if the repetitions is greater than 2, then the parser should read as many digits as possible to construct year. Then it will throw exception when parser meets 'M'.

Reply:
Let's maybe introduce a setting then and change the category to "backwards incompatible"? Some people may rely on current behavior

Reply:
Tbh, that sounds a bit strange. It means that it won't work without separators at all, leading to `yyyyMM` being an invalid pattern basically